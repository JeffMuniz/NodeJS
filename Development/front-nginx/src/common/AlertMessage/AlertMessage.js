import React from "react";
import { SvgIcon, Button } from "@common";
import { string, func, shape, number } from "prop-types";
import { If } from "@utils";
import { buttonTypes } from "@enums";
import {
  IconWrapper,
  Footer,
  Container,
  Title,
  Message,
} from "./AlertMessage.styles";

const AlertMessage = ({
  title,
  iconName,
  message,
  buttonAction,
  buttonText,
  btnProps,
  height,
  width,
  cancelBtnText,
  cancelBtnProps,
  cancelBtnAction,
}) => (
  <Container id="small_modal" width={width} height={height}>
    <IconWrapper iconName={iconName}>
      <SvgIcon name={iconName} />
    </IconWrapper>
    <Title iconName={iconName}>{title}</Title>
    <If test={message}>
      <Message>{message}</Message>
    </If>
    <If test={buttonText || cancelBtnText}>
      <Footer>
        <If test={cancelBtnText}>
          <Button
            id="action_btn_cancel"
            buttonType={buttonTypes.light}
            {...cancelBtnProps}
            value={cancelBtnText}
            onClick={cancelBtnAction}
          />
        </If>
        <If test={buttonText}>
          <Button
            id="action_btn"
            {...btnProps}
            value={buttonText}
            onClick={buttonAction}
          />
        </If>
      </Footer>
    </If>
  </Container>
);

AlertMessage.propTypes = {
  title: string.isRequired,
  iconName: string.isRequired,
  message: string,
  width: string,
  buttonAction: func,
  buttonText: string,
  btnProps: shape({}),
  height: number,
  cancelBtnText: string.isRequired,
  cancelBtnProps: shape({}),
  cancelBtnAction: func.isRequired,
};

AlertMessage.defaultProps = {
  width: "312px",
  message: undefined,
  buttonText: undefined,
  buttonAction: () => {},
  btnProps: {},
  height: null,
  cancelBtnProps: {},
};

export default AlertMessage;
