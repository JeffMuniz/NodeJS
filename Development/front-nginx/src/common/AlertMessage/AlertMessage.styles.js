import styled from "styled-components";
import { warningYellow, persianRed } from "@colors";

export const Container = styled.div`
  width: ${({ width }) => width};
  height: ${({ height }) => height || "auto"};
  padding: 34px 24px 24px 24px;
  display: flex;
  align-items: center;
  flex-direction: column;
`;

export const IconWrapper = styled.div`
  margin-bottom: ${props => (props.iconName === "error" ? "0" : "17px")};
`;

export const Title = styled.h1`
  color: ${props => (props.iconName === "error" ? persianRed : warningYellow)};
  font-style: normal;
  font-weight: bold;
  font-size: ${props => (props.iconName === "error" ? "20px" : "28px")};
  line-height: 37px;
  letter-spacing: 0.3px;
  margin: 0;
  margin-bottom: 12px;
`;

export const Message = styled.p`
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 26px;
  text-align: center;
  padding: 0;
`;

export const Footer = styled.div`
  margin-top: 22px;
  width: 100%;
  display: flex;
  justify-content: space-between;
`;
