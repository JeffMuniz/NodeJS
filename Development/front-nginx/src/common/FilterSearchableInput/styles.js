import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  justify-content: flex-end;
  margin-bottom: "40px";
  margin-left: auto;
  height: 40px;
  width: 100%;
`;
