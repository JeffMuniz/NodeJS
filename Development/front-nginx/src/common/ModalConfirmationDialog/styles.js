import styled from "styled-components";
import { Button as CommonButton } from "@common";

export const Button = styled(CommonButton)`
  height: 50px;
  min-width: 140px;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  margin-top: 30px;
`;

export const ContentModal = styled.div`
  text-align: center;
`;

export const Text = styled.div`
  margin-top: 15px;
`;
