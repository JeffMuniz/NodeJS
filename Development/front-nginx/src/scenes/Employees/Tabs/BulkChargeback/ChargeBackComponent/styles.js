import React from "react";
import styled from "styled-components";

import { white, blue, red, inputBorder, lighterBlack } from "@colors";
import { ifStyle } from "@utils";
import { SvgIcon, Button as BaseButton } from "@common";

const isActive = ifStyle("active");

export const Icon = styled(props => <SvgIcon {...props} />)`
  position: absolute;
  left: 0;
  margin: 7px;
`;

export const CustomInputWrap = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
  position: relative;
  width: 100%;
  margin-top: 0px;
`;

export const SelectWrap = styled.span`
  flex-direction: row;
  display: flex;
  top: 8px;
  right: 0;
  text-align: center;
  color: ${blue};
  min-width: 260px;
  white-space: nowrap;
`;

export const SelectOptions = styled.ul`
  position: absolute;
  top: 5px;
  right: -65px;
  width: 200px;
  padding: 0;
  background: ${white};
  box-shadow: 1px 4px 20px rgba(0, 0, 0, 0.2);
  border-radius: 4px;
  z-index: 5;
`;

export const SelectIcon = styled.div`
  cursor: pointer;
`;

export const StyledOption = styled.li`
  list-style: none;
  display: block;
  text-align: left;
  padding: 23px 0;
  border-bottom: 1px solid ${inputBorder};
  margin: 0 25px;
  cursor: pointer;
  font-weight: ${isActive("bold", "normal")};
  color: ${lighterBlack};
  &:last-child {
    border-bottom: none;
  }
`;

export const Text = styled.span`
  color: ${lighterBlack};
`;

export const Wrapper = styled.div`
  padding: 20px;
  width: 620px;
`;

export const ButtonsWrapper = styled.div`
  margin: 20px auto;
  display: flex;
  justify-content: center;
  max-width: 400px;
`;

export const FilterWrapper = styled.div`
  display: flex;
`;

export const ActionButtonsWrapper = styled.div`
  flex-grow: 1;
  display: flex;
  justify-content: flex-end;
  align-items: center;
`;

export const Button = styled(BaseButton)`
  color: ${red};
`;
