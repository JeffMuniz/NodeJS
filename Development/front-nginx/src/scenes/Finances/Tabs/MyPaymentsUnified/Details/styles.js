import styled from "styled-components";

import { blue, darkenGrey, rgbaMediumGrey, whiteF2 } from "@colors";
import { Button as CommonButton } from "@common";

export const Button = styled(CommonButton)`
  font-size: 13px;
  font-weight: 800;
  height: 30px;
  padding: 0;
`;

export const Container = styled.div`
  background: ${whiteF2};
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  line-height: 26px;
  margin-bottom: 30px;
  padding: 44px 20px 20px;
`;

export const ContainerButton = styled.div`
  position: absolute;
`;

export const ContainerOrders = styled.div`
  margin-top: 30px;
`;

export const Column = styled.div`
  min-width: ${props => props.minWidth};
  max-width: 400px;
  border-right: ${props =>
    props.withVerticalBorder && `1px solid ${darkenGrey}`};
  padding-right: ${props => props.withVerticalBorder && "55px"};
`;

export const DetailButton = styled.button`
  background: transparent;
  color: ${blue};
  cursor: pointer;
  font-size: ${props => props.fontSize || "14px"};
  font-weight: ${props => props.fontWeight || "600"};
  line-height: 16px;
  border: none;
`;

export const Item = styled.div`
  padding-right: 10px;
  margin-top: ${props => props.marginTop};
`;

export const Label = styled.div`
  display: ${props => (props.vertical ? "block" : "inline-block")};
  font-size: ${props => (props.highlighted ? "17px" : "14px")};
  font-weight: ${props => (props.boldLabel || props.highlighted) && "bold"};
  width: ${props => props.widthLabel && "135px"};
  min-height: ${props => (props.highlighted ? "50px" : "30px")};
`;

export const Value = styled.div`
  display: ${props => (props.vertical ? "block" : "inline-block")};
  font-size: ${props => (props.highlighted ? "16px" : "14px")};
  font-weight: ${props => props.boldValue && "bold"};
  margin-left: ${props => props.vertical || props.marginLeft || "5px"};
  color: ${props => props.color};
  min-height: 30px;
`;

export const Line = styled.div`
  border-top: 1px solid ${darkenGrey};
`;

export const LoadingWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-grow: 1;
  height: 100%;
  width: 100%;
`;

export const Row = styled.div`
  background-color: ${rgbaMediumGrey};
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
  margin-bottom: 8px;
  padding: 16px 25px;
`;
