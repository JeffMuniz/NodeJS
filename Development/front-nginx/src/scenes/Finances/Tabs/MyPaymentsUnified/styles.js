import styled from "styled-components";
import { rgbaMediumGrey, lighterBlack } from "@colors";

export const Table = styled.div`
  display: table;
  border-collapse: separate;
  border-spacing: 0 6px;
  width: 100%;
`;

export const InvoiceRow = styled.div`
  background-color: ${rgbaMediumGrey};
  display: table-row;
  margin-bottom: 8px;
  padding: 16px 25px;
`;

export const Field = styled.div`
  gap: 6px;
  align-items: center;
  display: ${props => props.isFlex && "-webkit-box"};
`;

export const Label = styled.div`
  line-height: 22px;
  font-size: 12px;
  color: ${lighterBlack};
`;

export const Value = styled.div`
  line-height: 26px;
  font-size: 14px;
  color: ${lighterBlack};
  font-weight: bold;
`;

export const FiltersWrapper = styled.div`
  display: flex;
  margin: 0 10px 20px;
`;

export const Cell = styled.div`
  display: table-cell;
  padding: 12px 18px;
  vertical-align: middle;
  border-top-left-radius: ${props => props.withRadiusLeft && "10px"};
  border-bottom-left-radius: ${props => props.withRadiusLeft && "10px"};
  border-top-right-radius: ${props => props.withRadiusRight && "10px"};
  border-bottom-right-radius: ${props => props.withRadiusRight && "10px"};
`;

export const CellWithBorder = styled.div`
  border-right: 1px solid #9b9b9b4c;
  border-left: 1px solid #9b9b9b4c;
  padding-left: 34px;
`;
