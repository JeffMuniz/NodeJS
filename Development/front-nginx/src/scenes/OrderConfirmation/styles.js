import styled from "styled-components";
import { Button as CommonButton } from "@common";

import { blue } from "@colors";

export const Button = styled(CommonButton)`
  height: 50px;
  width: ${props => props.width || "330px"};
`;

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  margin-top: 30px;
`;

export const Content = styled.div`
  margin: 0 90px;
`;

export const ContentBody = styled.div`
  display: flex;
`;

export const ContentModal = styled.div`
  text-align: center;
`;

export const ErrorMessage = styled.div`
  display: flex;
  justify-content: center;
  color: red;
  padding: 20px;
`;

export const Field = styled.div`
  display: flex;
  justify-content: space-between;
  line-height: 30px;
  font-size: 14px;
  border-bottom: ${props =>
    props.isDotted ? ".15rem dotted" : props.withDivisor && "1px solid"};
  border-bottom-color: #ccc;
  margin-bottom: ${props =>
    props.addSeparationArea
      ? "40px"
      : props.withDivisor && props.isDotted && "10px"};
`;

export const FieldLabel = styled.div`
  font-weight: ${props => (props.isBold || props.isLink) && "bold"};
  color: ${props => props.isLink && blue};
  cursor: ${props => props.isLink && "pointer"};
`;

export const FieldValue = styled.div`
  font-size: 15px;
  font-weight: ${props => props.isBold && "bold"};
`;

export const Footer = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 30px;
`;

export const PaymentCard = styled.div`
  background-color: white;
  height: 70px;
  border-radius: 0.7rem;
  box-shadow: 1px 4px 4px rgba(0, 0, 0, 0.3);
  border: 1px solid #ccc;
  display: flex;
  align-items: center;
  padding-left: 20px;
  margin-bottom: 15px;
`;

export const PaymentModes = styled.div`
  flex: 1;
`;

export const SubTitle = styled.div`
  font-size: 16px;
  font-weight: bold;
  margin-bottom: 15px;
`;

export const Summary = styled.div`
  flex: 1;
  padding-right: 150px;
`;

export const Text = styled.div`
  margin-top: 15px;
`;
