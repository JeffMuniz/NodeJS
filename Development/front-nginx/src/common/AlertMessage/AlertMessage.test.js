import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import TestRenderer from "react-test-renderer";
import { Message, Footer } from "./AlertMessage.styles";

import AlertMessage from "./AlertMessage";

jest.mock("@common", () => ({
  SvgIcon: "SvgIcon",
  Button: "Button",
}));

const defaultProps = {
  title: "Lorem",
  iconName: "Ipsum",
};

describe("AlertMessage", () => {
  it("Should render AlertMessage", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<AlertMessage {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });

  it("Should render Message if a message prop is present", () => {
    // given
    const testRenderer = TestRenderer.create(
      <AlertMessage {...defaultProps} message="Hello" />,
    );
    const instance = testRenderer.root;
    const message = instance.findAllByType(Message);

    // then
    expect(message).toHaveLength(1);
  });

  it("Should not render Message if a message prop is not present", () => {
    // given
    const testRenderer = TestRenderer.create(
      <AlertMessage {...defaultProps} />,
    );
    const instance = testRenderer.root;
    const message = instance.findAllByType(Message);

    // then
    expect(message).toHaveLength(0);
  });

  it("Should render Footer if a buttonText prop is present", () => {
    // given
    const testRenderer = TestRenderer.create(
      <AlertMessage {...defaultProps} buttonText="Click me" />,
    );
    const instance = testRenderer.root;
    const footer = instance.findAllByType(Footer);

    // then
    expect(footer).toHaveLength(1);
  });

  it("Should not render Footer if a buttonText prop is not present", () => {
    // given
    const testRenderer = TestRenderer.create(
      <AlertMessage {...defaultProps} />,
    );
    const instance = testRenderer.root;
    const footer = instance.findAllByType(Footer);

    // then
    expect(footer).toHaveLength(0);
  });
});
