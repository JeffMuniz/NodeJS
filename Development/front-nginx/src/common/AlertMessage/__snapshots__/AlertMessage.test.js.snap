// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertMessage Should render AlertMessage 1`] = `
<styled.div
  height={null}
  id="small_modal"
  width="312px"
>
  <styled.div
    iconName="Ipsum"
  >
    <SvgIcon
      name="Ipsum"
    />
  </styled.div>
  <styled.h1
    iconName="Ipsum"
  >
    Lorem
  </styled.h1>
  <If
    test={undefined}
  >
    <styled.p />
  </If>
  <If
    test={undefined}
  >
    <styled.div>
      <If
        test={undefined}
      >
        <Button
          buttonType="light"
          id="action_btn_cancel"
          onClick={undefined}
          value={undefined}
        />
      </If>
      <If
        test={undefined}
      >
        <Button
          id="action_btn"
          onClick={[Function]}
          value={undefined}
        />
      </If>
    </styled.div>
  </If>
</styled.div>
`;
